components/script/dom/bindings/typedarrays.rs

```diff
@@ -142,3 +161,28 @@ where
         TypedArray::from(dest.get())
     }
 }
+
+pub fn create_new_external_array_buffer(
+    cx: JSContext,
+    mapping: Arc<Mutex<Vec<u8>>>,
+    offset: usize,
+    range_size: usize,
+    m_end: usize,
+) -> *mut JSObject {
```

Comment:
So, I'm not completely sure if exposing `pub fn new` on `HeapTypedArray` is a good idea, and we don't need to make that decision yet, because we can simply return `HeapTypedArray::<ArrayBufferU8>` from this function, so let's do that for now and see how the public API of the heapd typed array further evolves in subsequent PRs.

Reply:
As a quick update: another reason for not returning the pointer is that we should mark the function itself as unsafe if we did that, since it would be possible for someone to do anything with the pointer before having passed it to `new`.

Reply:
Indeed I will return `HeapTypedArray<T>`